addons/mail/static/tests/chatter/web/chatter_topbar.test.js

```diff
@@ -138,38 +139,40 @@ test("attachment counter with attachments", async () => {
 });
 
 test("attachment counter while loading attachments", async () => {
+    const def = new Deferred();
     const pyEnv = await startServer();
     const partnerId = pyEnv["res.partner"].create({});
-    onRpc("/mail/data", async (request) => {
-        const { params } = await request.json();
-        if (params.fetch_params.some((fetchParam) => fetchParam[0] === "mail.thread")) {
-            await new Deferred(); // simulate long loading
-        }
+    const { waitStoreFetch } = listenStoreFetch("mail.thread", async () => {
+        asyncStep("before mail.thread");
```

Comment:
Note: this step is necessary to ensure the callback is actually applied, it makes the test much more robust.